### PR TITLE
LPS-132150 Fix error registering GraphQLDTOContributors because Facet GraphQLObjectType was not registered at that moment

### PR DESCRIPTION
**Pull request by javierdearcos**
The bug was caused by a race condition because types coming from annotations are loaded asynchronously and when the type was requested by the GraphQLDTODefinition to be used as part of the Page definition, it was not available yet. To solve it I try to check if the type is available and if not, register it manually.

This solution causes a conflict when the Facet class GraphQL annotations were processed and try to register the type again, so to avoid this conflict I choose to remove the annotations and always register the Facet type at the beginning to be available for GraphQL annotations processor and for the GraphQLDTOContributor registry.

I think it could be a good approach as we mentioned several times that we wanted to remove the GraphQL annotations lib.

I left both commits so you can check the process but we can squash them before sending the PR

### CODE REVIEW

**Comment by javierdearcos** [comment]
Seems there is a missing baseline in the Failures in common with acceptance upstream results at 6e25ad1. Adding it

**Comment by nhpatt** [comment]
> The bug was caused by a race condition because types coming from annotations are loaded asynchronously

Are you sure that's the case?

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/graphql/servlet/GraphQLServletExtender.java [hunk 1]
```diff
@@ -1237,6 +1256,11 @@ private Servlet _createServlet() throws Exception {
 			Map<Class<?>, Set<Class<?>>> classesMap =
 				processingElementsContainer.getExtensionsTypeRegistry();
 
+			processingElementsContainer.setTypeRegistry(
```
Comment (nhpatt) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/graphql/servlet/GraphQLServletExtender.java:1259]: Why here? looks like more of a field related thing that should be before _collectObjectFields

**Review by nhpatt — COMMENTED** [review]
Code is Ok. But I'm worried about the bug... can you double check if 1) you can reproduce it several times (I can't) 2) check if the annotations are really processed asynchronously. Thanks!

**Comment by javierdearcos** [comment]
I've tried to reproduce it all afternoon without success 😞 

I'm pretty sure that the other day the classes were registered in an asynchronous way because as it executes the register... methods, the registered classes increase by hundreds (when there weren't so many registered) and the Facet class wasn't present, although it should be there just after the collect... methods.

Maybe I was wrong the other day and it is better to close the bug until it happens again if it does

**Comment by nhpatt** [comment]
I've checked the code internally and I don't see where it could be asynchronous... the error makes sense if there were no ServletData registered but having APIs in the server it does not makes sense.

Closing for now, if it happens again, we'll reopen.

**Comment by javierdearcos** [comment]
Ok, I'll close the bug as no longer reproducible too

**Pull request closed without merging**